cluster/engine.go

```diff
@@ -521,17 +521,17 @@ func (e *Engine) Create(config *ContainerConfig, name string, pullImage bool) (*
 	dockerConfig.CpuShares = int64(math.Ceil(float64(config.CpuShares*1024) / float64(e.Cpus)))
 	dockerConfig.HostConfig.CpuShares = dockerConfig.CpuShares
 
-	if id, err = client.CreateContainer(&dockerConfig, name); err != nil {
+	if id, err = client.CreateContainer(&dockerConfig, name, nil); err != nil {
 		// If the error is other than not found, abort immediately.
 		if err != dockerclient.ErrImageNotFound || !pullImage {
 			return nil, err
 		}
 		// Otherwise, try to pull the image...
-		if err = e.Pull(config.Image, nil); err != nil {
+		if err = e.Pull(config.Image, authConfig); err != nil {
 			return nil, err
 		}
 		// ...And try again.
-		if id, err = client.CreateContainer(&dockerConfig, name); err != nil {
+		if id, err = client.CreateContainer(&dockerConfig, name, nil); err != nil {
 			return nil, err
```

Comment:
The engine itself currently does not pull on CreateContainer API calls.  Swarm is the only place where this logic is implemented.

The reason I updated dockerclient was to make sure that when the incoming AuthConfig is unmarshaled at https://github.com/docker/swarm/pull/1384/files#diff-379151c15066f365334797e959e6ea94R435 that the registrytoken isn't dropped.  I suppose an alternative would have been to pass the base64 encoded header around, but that felt like a bigger deviation from the current flow.